exchanges/binance/binance_test.go

```diff
@@ -1965,6 +1968,31 @@ func TestGetDepositAddress(t *testing.T) {
 	}
 }
 
+func BenchmarkWsHandleData(bb *testing.B) {
+	bb.ReportAllocs()
+	ap, err := b.CurrencyPairs.GetPairs(asset.Spot, false)
+	require.NoError(bb, err)
+	err = b.CurrencyPairs.StorePairs(asset.Spot, ap, true)
+	require.NoError(bb, err)
+	go func() {
+		for {
+			<-b.Websocket.DataHandler
+		}
+	}()
+	bb.ResetTimer()
+	for i := 0; i < bb.N; i++ {
+		fixture, err := os.Open("testdata/wsHandleData.json")
```

Comment:
```
func BenchmarkWsHandleData(bb *testing.B) {
	bb.ReportAllocs()
	ap, err := b.CurrencyPairs.GetPairs(asset.Spot, false)
	require.NoError(bb, err)
	err = b.CurrencyPairs.StorePairs(asset.Spot, ap, true)
	require.NoError(bb, err)

	data, err := os.ReadFile("testdata/wsHandleData.json")
	require.NoError(bb, err)
	lines := bytes.Split(data, []byte("\n"))
	require.Len(bb, lines, 10)

	go func() {
		for {
			<-b.Websocket.DataHandler
		}
	}()

	bb.ResetTimer()
	for i := 0; i < bb.N; i++ {
		for x := range lines {
			assert.NoError(bb, b.wsHandleData(lines[x]))
		}
	}
}
```

Can move the setup above the benchmark iterations, since this is currently opening and closing this file thousands to tens of thousands of times per benchmark tes 👀

Reply:
Much better, thanks! Didn't really impact benchmark performance, but much better than what I was doing 😄